python/tests/stress/arcticdb/test_mem_leaks.py

```diff
@@ -0,0 +1,190 @@
+
+        assert max_total_mem_lost_threshold_bytes >= process_growth , f"Memory of the process grew more than defined threshold: {nice_bytes_str(process_growth)} (specified: {nice_bytes_str(max_total_mem_lost_threshold_bytes)} )"
+        assert max_machine_memory_percentage >= mem_per , f"Machine utilized more memory than specified threshold :{mem_per}% (specified {max_machine_memory_percentage}%)"
+        
+    print ("The process assessment finished within expectations. Total consumed additional mem is bellow threshold: ", process_growth)
+    print ("Returning this number as result.")
+    return process_growth, count_drops(mem_growth_each_iter, 0.5)
+
+def grow_exp(df_to_grow : pd.DataFrame, num_times_xx2:int):
+    """
+        Quickly inflates the dataframe by doubling its size for each iteration
+        Thus the final size will be 'num_times_xx2' power of two
+    """
+    for n in range(1,num_times_xx2):
+        df_prev = df_to_grow.copy(deep=True)
+        df_to_grow = pd.concat([df_to_grow, df_prev])
+    return df_to_grow
+
+def test_mem_leak_read_all_arctic_lib(arctic_library_lmdb):
+    lib : adb.Library = arctic_library_lmdb
+
+    df = get_sample_dataframe(size=1000000)
+    df = grow_exp(df, 5)
+
+    symbol = "test"
+    lib.write(symbol, df)
+
+    def proc_to_examine():
+        data = lib.read(symbol).data
+        del data
+
+    df = lib.read(symbol).data
+    print(df)
+    del df
+
+    '''
+        To determine optimal max_mem first set the procedure to a very high number like 10 GB
+        and run it from commandline for approx 10 iteration.
+
+        > python -m pytest -s python/tests/stress/arcticdb/test_mem_leaks.py::test_mem_leak_read_all_arctic_lib
+
+        Observe 2 things:
+         - maximum process memory growth
+         - ideally there should be one iteration the mem did go down indicating GC in C++ code
+
+         if both conditions are met then the process is likely to not leak big chunks of mem
+         and you can take the max mem growth number add 20 percent safety margin and you will
+         arrive at reasonable  max_mem
+
+         Then put for number of iterations 3x or 5x and this mem limit and 
+         run the test from command line again to assure it runs ok before commit 
+
+    '''
+    max_mem_bytes = 153623040
+
+    check_process_memory_leaks(proc_to_examine, 20, max_mem_bytes, 80.0)
+
+def test_mem_leak_read_all_native_store(lmdb_version_store_very_big_map):
+    lib : NativeVersionStore = lmdb_version_store_very_big_map
+
+    df = get_sample_dataframe(size=1000000)
+    df = grow_exp(df, 5)
+
+    symbol = "test"
+    lib.write(symbol, df)
+
+    def proc_to_examine():
+        data = lib.read(symbol).data
+        del data
+
+    df = lib.read(symbol).data
+    print(df)
+    del df
+
+    """ 
+        See comment in previous test
+    """
+    max_mem_bytes = 608662528
```

Comment:
Surely we can have a tighter threshold than 600GB!

Reply:
one execution was 430 MB another little over 100MB. Not sure It is good idea to optimized more see my previous comment

Reply:
So it is 600MB the limit here....

Reply:
Maths fail from me :)